components/team_controller/team_controller.tsx

```diff
@@ -0,0 +1,258 @@
+            iNoBounce.enable();
+        }
+
+        // Set up tracking for whether the window is active
+        window.isActive = true;
+
+        LocalStorageStore.setTeamIdJoinedOnLoad(null);
+
+        return () => {
+            window.isActive = false;
+
+            if (browserIsIosSafari) {
+                iNoBounce.disable();
+            }
+        };
+    }, []);
+
+    async function initTeamOrRedirect(team: Team) {
+        try {
+            await props.initializeTeam(team);
+            setTeam(team);
+        } catch (error) {
+            history.push('/error?type=team_not_found');
+        }
+    }
+
+    async function joinTeamOrRedirect(teamNameParam: string, joinedOnFirstLoad: boolean) {
+        try {
+            // skip reserved team names
+            if (Constants.RESERVED_TEAM_NAMES.includes(teamNameParam)) {
+                throw new Error('Team name is reserved');
+            }
+
+            const {data: joinedTeam} = await props.joinTeam(teamNameParam, joinedOnFirstLoad) as ActionResult<Team, ServerError>; // Fix in MM-46907;
+            if (joinedTeam) {
+                setTeam(joinedTeam);
+            } else {
+                throw new Error('Unable to join team');
+            }
+        } catch (error) {
+            history.push('/error?type=team_not_found');
+        }
+    }
+
+    const teamsListDependency = props.teamsList.map((team) => team.id).sort().join('+');
+
+    // Effect to run when url for team or teamsList changes
+    useEffect(() => {
+        // Prevents the RHS from closing when clicking on a global permalink.
+        emitCloseRightHandSide();
+
+        const teamFromURLParam = getTeamFromTeamList(props.teamsList, teamNameParam);
+        if (teamFromURLParam) {
+            // Team switch
+            initTeamOrRedirect(teamFromURLParam);
+            return;
+        }
+
+        if (teamNameParam && !team) {
+            // Team join on first load
+            joinTeamOrRedirect(teamNameParam, true);
+            return;
+        }
+
+        if (teamNameParam && team && team.name !== teamNameParam) {
+            // Team join by going through a link
+            joinTeamOrRedirect(teamNameParam, false);
+        }
+    }, [teamNameParam, teamsListDependency]);
+
+    if (props.mfaRequired) {
+        history.push('/mfa/setup');
+        return null;
+    }
+
+    if (team === null) {
+        return null;
+    }
+
+    return (
+        <Switch>
+            <Route
+                path={'/:team/integrations'}
+                component={BackstageController}
+            />
+            <Route
+                path={'/:team/emoji'}
+                component={BackstageController}
+            />
+            {props.plugins?.map((plugin) => (
+                <Route
+                    key={plugin.id}
+                    path={'/:team/' + (plugin as any).route}
+                    render={() => (
```

Comment:
Not sure what are the implications of `Route` being passed new a new `render` function on every render, but it is recommended not to pass anonymous functions to the props of a render. Use a `useCallback` instead.

Reply:
Will check this later https://mattermost.atlassian.net/browse/MM-48261 . As this is as it was before.